DiagnosticsTests/Reporters/SmartInsights/SmartInsightsReporterTests.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  SmartInsightsReporterTests.swift
+//  
+//
+//  Created by Antoine van der Lee on 10/02/2022.
+//
+
+import XCTest
+@testable import Diagnostics
+
+final class SmartInsightsReporterTests: XCTestCase {
+
+    func testSmartInsightsChapter() throws {
+        let reporter = SmartInsightsReporter()
+        let chapter = reporter.report()
+        XCTAssertEqual(chapter.title, "Smart Insights")
+        let insightsDictionary = try XCTUnwrap(chapter.diagnostics as? [String: String])
+        XCTAssertFalse(insightsDictionary.isEmpty)
+    }
+    
+    func testRemovingDuplicateInsights() throws {
+        var reporter = SmartInsightsReporter()
+        let insight = SmartInsight(name: UUID().uuidString, result: .success(message: UUID().uuidString))
+        
```

Comment:
<!--
  0 failure: 
  1 warning:  Lines should not ...
  
  
  DangerID: danger-id-Danger;
  File: DiagnosticsTests/Reporters/SmartInsights/SmartInsightsReporterTests.swift;
  Line: 24;
-->

- :warning: Lines should not have trailing whitespace. (`trailing_whitespace`)